extensions/android-paging3-rx3/src/main/java/com/squareup/sqldelight/android/paging3/RxOffsetQueryPagingSource.kt

```diff
@@ -0,0 +1,59 @@
+package com.squareup.sqldelight.android.paging3
+
+import androidx.paging.PagingState
+import androidx.paging.rxjava3.RxPagingSource
+import com.squareup.sqldelight.Query
+import com.squareup.sqldelight.Transacter
+import io.reactivex.rxjava3.core.Scheduler
+import io.reactivex.rxjava3.core.Single
+import kotlinx.coroutines.rx3.asCoroutineDispatcher
+import kotlinx.coroutines.rx3.rxSingle
+
+internal class RxOffsetQueryPagingSource<RowType : Any>(
+  queryProvider: (limit: Long, offset: Long) -> Query<RowType>,
+  countQuery: Query<Long>,
+  transacter: Transacter,
+  scheduler: Scheduler,
+) : RxPagingSource<Long, RowType>() {
+
+  private val delegate = QueryPagingSource(
+    countQuery = countQuery,
+    transacter = transacter,
+    queryProvider = queryProvider,
+    dispatcher = scheduler.asCoroutineDispatcher(),
+  )
+
+  override fun getRefreshKey(state: PagingState<Long, RowType>) = delegate.getRefreshKey(state)
+
+  override fun loadSingle(params: LoadParams<Long>): Single<LoadResult<Long, RowType>> = rxSingle {
```

Comment:
Since this class delegates to `OffsetQueryPagingSource`...do we need tests?

Reply:
Cursory (heh) tests for functionality are fine. Don't need to exhaustively test the whole impl.

Reply:
Added a single dataset exhaustion test